packages/eslint-plugin-ckeditor5-rules/lib/rules/license-header.js

```diff
@@ -32,75 +33,109 @@ module.exports = {
-
-				if ( isLeadingCommentALicenseComment ) {
-					const leadingCommentText = sourceCode.getText( leadingComment );
-
-					if ( leadingCommentText !== headerText ) {
-						context.report( {
-							node: leadingComment,
-							message: 'The license header is incorrect.'
-						} );
-					}
+			Program() {
+				licenseHeaderRule( context, headerLines );
+			}
+		};
+	}
+};
 
-					if ( !leadingShebang && leadingComment.loc.start.line > 1 ) {
-						context.report( {
-							loc: {
-								start: {
-									line: 0,
-									column: 0
-								},
-								end: {
-									line: 0,
-									column: 1
-								}
-							},
-							message: 'There is an extra new line before the license header.'
-						} );
-					}
+function licenseHeaderRule( context, headerLines ) {
+	const headerText = headerLines.join( '\n' );
+	const sourceCode = context.getSourceCode();
+	const fullSourceString = sourceCode.lines.join( '\n' );
 
-					if ( sourceCode.lines[ leadingComment.loc.end.line ] != '' ) {
-						context.report( {
-							loc: {
-								start: {
-									line: headerHeight,
-									column: 0
-								},
-								end: {
-									line: headerHeight,
-									column: 0
-								}
-							},
-							message: 'There is a new line missing after the license header.'
-						} );
-					}
-				} else {
-					context.report( {
-						loc: {
-							start: {
-								line: 0,
-								column: 0
-							},
-							end: {
-								line: 0,
-								column: 0
-							}
-						},
-						message: 'The license header is missing.'
-					} );
+	const comments = sourceCode.getAllComments();
+	const shebang = comments.find( comment => comment.type === 'Shebang' );
+	const licenseComment = comments.find( comment => comment.type === 'Block' && comment.value.toLowerCase().includes( '@license' ) );
+
+	if ( !licenseComment ) {
+		const line = shebang ? 2 : 0;
+
+		context.report( {
+			loc: {
+				start: {
+					line,
+					column: 0
+				},
+				end: {
+					line,
+					column: 0
 				}
+			},
+			message: 'The license header is missing.',
+			fix: fixer => {
+				return shebang ?
+					fixer.insertTextAfter( shebang, '\n\n' + headerText + '\n\n' ) :
+					fixer.insertTextAfterRange( [ 0, 0 ], headerText + '\n\n' );
 			}
-		};
+		} );
+
+		// If there is no license header or we just created it, there is no need for other checks.
+		return;
 	}
-};
+
+	if ( sourceCode.getText( licenseComment ) !== headerText ) {
+		context.report( {
+			node: licenseComment,
+			message: 'The license header is incorrect.',
+			fix: fixer => fixer.replaceTextRange( licenseComment.range, headerText )
+		} );
+	}
+
+	const contentBeforeStart = shebang ? shebang.range[ 1 ] : 0;
+	const contentBeforeRange = [ contentBeforeStart, licenseComment.range[ 0 ] ];
+	const contentBefore = fullSourceString.slice( ...contentBeforeRange );
+	const whitespaceOnly = /^\s+$/.test( contentBefore );
+	const expectedContentBefore = shebang ? '\n\n' : '';
+
+	const ContentBeforeLoc = {
```

Comment:
Is there a specific reason that a variable name is starting with uppercase?

Reply:
No, it was a mistake, thanks for catching this.